packages/ui-utils/src/types.ts

```diff
@@ -117,14 +117,26 @@ export type ToolInvocation =
   | CoreToolCall<string, any>
   | CoreToolResult<string, any, any>;
 
+export interface TextContentPart {
+  type: 'text';
+  text: string;
+}
+
+export interface ImageContentPart {
+  type: 'image';
+  image: string;
+}
+
+export type ContentPart = TextContentPart | ImageContentPart;
+
 /**
  * AI SDK UI Messages. They are used in the client and to communicate between the frontend and the API routes.
  */
 export interface Message {
   id: string;
   createdAt?: Date;
 
-  content: string;
+  content: string | ContentPart[] | null | undefined;
```

Comment:
alternatively we could introduce `parts` here and leave `content` as is for backwards compat

Reply:
done!